src/codegen/lang/if.cpp

```diff
@@ -42,6 +42,27 @@ If::If(CodeGen &cg, llvm::Value *if_condition, const std::string &name)
 If::If(CodeGen &cg, const codegen::Value &if_condition, const std::string &name)
     : If(cg, type::Boolean::Instance().Reify(cg, if_condition), name) {}
 
+llvm::Value *If::If2(CodeGen &cg, llvm::Value *if_condition, 
+                   const std::function<llvm::Value *()> &cgen_then_branch,
+                   const std::function<llvm::Value *()> &cgen_else_branch) {
+  auto new_if = new If(cg, if_condition, "new_if");
+
+  // Run the then-branch
+  llvm::Value *then_result = cgen_then_branch();
+  new_if->ElseBlock();
+  // Run the else-branch
+  llvm::Value *else_result = cgen_else_branch();
+  new_if->EndIf();
+  return new_if->BuildPHI(then_result,else_result);
+}
+
+llvm::Value *If::If2(CodeGen &cg, const codegen::Value &if_condition, 
```

Comment:
I don't think we should make a new `If::If2`. If we want to go this way, we should just replace the old `If::If` since having two of them will confuse people.

Reply:
The old `If::If` is a constructor which can only give a `codegen::lang::If` type variable instead of `llvm::Value *` type variable. If we want a `If` to return `llvm::Value *` directly as the issue posted, what if we make the `If::If2` less confusing by giving it a less confusing name like `If::GetIfResult` or something like that. Or, maybe we should implement this whole class completely different by makes `codegen::lang::If` be a subclass of `llvm::Value *`.